Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -429,14 +436,10 @@ public class CarPlayNavigationViewController: UIViewController {
         let feedbackButtonHandler: (_ : CPGridButton) -> Void = { [weak self] (button) in
             self?.carInterfaceController.popTemplate(animated: true)
             
-            // TODO: Fix this Demeter violation with proper encapsulation
-            guard let uuid = self?.navigationService.eventsManager.recordFeedback() else { return }
```

Comment:
Repeated weak self usage can lead to race conditions. If the logic can't be completed without self it is better to guard to strong self at the beginning of the block:
```swift
guard let self = self else { /* optional logic */; return }
```